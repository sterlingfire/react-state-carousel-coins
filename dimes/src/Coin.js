import "./Coin.css";

const SIDE_TO_URL = {
  Heads: "https://upload.wikimedia.org/wikipedia/commons/d/dd/2017-D_Roosevelt_dime_obverse_transparent.png",
  Tails: "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d9/2017-D_Roosevelt_dime_reverse_transparent.png/220px-2017-D_Roosevelt_dime_reverse_transparent.png"
} 

/** Renders Coin component
 * 
 * Prop:
 * - Side: "Heads", "Tails" or "" which determines the image
 *   - "" does not display the image
 */

function Coin({side}) {
  let imgUrl = SIDE_TO_URL[side];

  return (<div>
    { side && <img src={imgUrl} className="Coin" alt={side}></img> }
  </div>);
}

export default Coin;
export {SIDE_TO_URL};