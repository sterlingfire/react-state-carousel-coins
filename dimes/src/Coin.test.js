import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Coin from "./Coin.js"


it("renders without error", function() {
  render(<Coin side="Heads" />);
});

it("matches snapshot", function() {
  const { container } = render(<Coin side="Heads"/>);
  expect(container).toMatchSnapshot();
});
