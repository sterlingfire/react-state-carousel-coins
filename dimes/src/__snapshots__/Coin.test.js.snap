// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <img
      alt="Heads"
      class="Coin"
      src="https://upload.wikimedia.org/wikipedia/commons/d/dd/2017-D_Roosevelt_dime_obverse_transparent.png"
    />
  </div>
</div>
`;
